### PR TITLE
automatic java17 profile https://github.com/ChronicleEnterprise/RootP…

### PR DESCRIPTION
**Pull request by JerryShea**
…om/issues/218

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -328,14 +331,10 @@
...
-                                --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
-                                --add-opens=jdk.compiler/com.sun.tools.javac=ALL-UNNAMED
-                                --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
-                                --add-opens=java.base/java.lang=ALL-UNNAMED
-                                --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
+                                --add-exports java.base/jdk.internal.ref=ALL-UNNAMED --add-exports=java.base/sun.nio.ch=ALL-UNNAMED
+                                --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
+                                --add-opens=jdk.compiler/com.sun.tools.javac=ALL-UNNAMED --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
+                                --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
                                 --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED
```
Comment (nicktindall) [inline pom.xml]: I bet we could achive this without having to duplicate all this stuff by using https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#systemPropertyVariables

Comment (minborg) [reply pom.xml]: These are not system properties but command line args to `java`. But perhaps I misunderstood your comment?

Comment (nicktindall) [reply pom.xml]: They're all added in the root parent pom. The only reason we duplicate them here is to add the -Duser.timezone=UTC parameter to argLine. If we use the system property configuration we don't need to duplicate the argLine as it should be inherited?

File: pom.xml [hunk 2]
```diff
@@ -306,43 +308,6 @@
...
-                                --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
-                                --add-opens=java.base/java.lang=ALL-UNNAMED
-                                --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
-                                --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED
-                            </argLine>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
```
Comment (nicktindall) [inline pom.xml:345]: Don't need this anymore as we're no longer overriding `argLine`

Comment (JerryShea) [reply pom.xml:345]: wow! thanks

**Review by nicktindall — APPROVED** [review]

**Pull request merged**